components/scripts/gradle/gradle-init-scripts/configure-remote-build-caching.gradle

```diff
@@ -4,23 +4,110 @@ static getInputParam(Gradle gradle, String name) {
     return gradle.startParameter.systemPropertiesArgs[name] ?: System.getProperty(name) ?: System.getenv(envVarName)
 }
 
+def isTopLevelBuild = !gradle.parent
+
+def expDir = getInputParam(gradle, 'develocity.build-validation.expDir')
+def remoteBuildCacheType = getInputParam(gradle, 'develocity.build-validation.remoteBuildCacheType')
 def remoteBuildCacheUrl = getInputParam(gradle, 'develocity.build-validation.remoteBuildCacheUrl')
 
-settingsEvaluated { settings ->
+def docsRoot = 'https://docs.gradle.com/develocity/gradle-plugin'
+
+settingsEvaluated { Settings settings ->
```

Comment:
There is now a lot of code in this block, and a lot of nested if statements. I think it would be helpful to extract some of the nested if statements into helper functions so that the code is overall easier to read.

Reply:
@jthurne I didn't necessarily address the nesting, but I significantly reduced the cognitive overload by extracting the error handling. The core logic first entirely on my screen which was not the case before. Do you think it's sufficient? See a8b7e1e693f000bcc17ad4eeef67c475c15e2ba8

Reply:
That is a significant improvement 👍